File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/AmountFunction.java
package cz.fb.manaus.reactor.profit.progress.function;

import cz.fb.manaus.core.model.SettledBet;
import org.springframework.stereotype.Component;

import java.util.OptionalDouble;

@Component
public class AmountFunction implements ProgressFunction {

    @Override
    public OptionalDouble apply(SettledBet bet) {
        return OptionalDouble.of(bet.getPrice().getAmount());
    }

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/AmountFunction.kt
package cz.fb.manaus.reactor.profit.progress.function

import cz.fb.manaus.core.model.RealizedBet
import org.springframework.stereotype.Component

@Component
object AmountFunction : ProgressFunction {

    override fun invoke(bet: RealizedBet): Double {
        return bet.settledBet.price.amount
    }

}
